packages/edit-site/src/components/dataviews/in-filter.js

```diff
@@ -2,62 +2,100 @@
  * WordPress dependencies
  */
 import {
-	__experimentalInputControlPrefixWrapper as InputControlPrefixWrapper,
-	SelectControl,
+	Button,
+	privateApis as componentsPrivateApis,
+	Icon,
 } from '@wordpress/components';
+import { check, chevronDown } from '@wordpress/icons';
 import { __, sprintf } from '@wordpress/i18n';
 
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
 export const OPERATOR_IN = 'in';
 
+const { DropdownMenuV2, DropdownMenuItemV2, DropdownMenuSeparatorV2 } = unlock(
+	componentsPrivateApis
+);
+
 export default ( { filter, view, onChangeView } ) => {
-	const valueFound = view.filters.find(
-		( f ) => f.field === filter.field && f.operator === OPERATOR_IN
+	const filterInView = view.filters.find( ( f ) => f.field === filter.field );
+	const activeElement = filter.elements.find(
+		( element ) => element.value === filterInView?.value
 	);
 
-	const activeValue =
-		! valueFound || ! valueFound.hasOwnProperty( 'value' )
-			? ''
-			: valueFound.value;
-
-	const id = `dataviews__filters-in-${ filter.field }`;
-
 	return (
-		<SelectControl
-			id={ id }
-			__nextHasNoMarginBottom
-			value={ activeValue }
-			prefix={
-				<InputControlPrefixWrapper
-					as="label"
-					htmlFor={ id }
-					className="dataviews__select-control-prefix"
-				>
-					{ sprintf(
-						/* translators: filter name. */
-						__( '%s:' ),
-						filter.name
-					) }
-				</InputControlPrefixWrapper>
+		<DropdownMenuV2
+			key={ filter.field }
+			label={ filter.name }
+			trigger={
+				<Button variant="tertiary" size="compact">
+					{ activeElement !== undefined
+						? sprintf(
+								/* translators: 1: Filter name. 2: filter value. e.g.: "Author is Admin". */
+								__( '%1$s is %2$s' ),
+								filter.name,
+								activeElement.label
+						  )
+						: filter.name }
+					<Icon icon={ chevronDown } />
```

Comment:
Have you tried using the `icon` prop on the `Button` component? Would that make any difference?

Reply:
Something like this could work:

`<Button icon={chevronDown} iconPosition="right" text={...} />`

Although:
- there seems to be a bug in the button component in recognizing where the button is icon-only or has text too
- I don't like the "right" keyword ("inline-end" would be better)
- in general, I prefer composition using `children`

So, I think that the current approach is better.